### PR TITLE
Remove lib.d.ts line info in inlay hint baselines

### PR DESCRIPTION
**Pull request by MariaSolOs**
Had some issues in #55141 because of including line information from `lib.d.ts` files, so this PR attempts to make those baselines more robust.

### CODE REVIEW

**Comment by jakebailey** [comment]
No test change; is this because your other PR is what introduces them? Was hoping to see an example that shows it working but otherwise it's probably fine.

File: src/harness/fourslashImpl.ts [hunk 1]
```diff
@@ -839,6 +839,15 @@ export class TestState {
         const fileName = this.activeFile.fileName;
         const hints = this.languageService.provideInlayHints(fileName, span, preferences);
         const annotations = ts.map(hints.sort(sortHints), hint => {
+            if (hint.displayParts) {
+                hint.displayParts = ts.map(hint.displayParts, part => {
+                    if (part.file && /lib(?:.*)\.d\.ts$/.test(part.file)) {
+                        part.span!.start = -1;
```
Comment (jakebailey) [inline src/harness/fourslashImpl.ts:845]: `span` is definitely defined?

**Comment by MariaSolOs** [comment]
> No test change; is this because your other PR is what introduces them? Was hoping to see an example that shows it working but otherwise it's probably fine.

Yep. My type hints PR will have the changes. I already tried cherry picking this locally and it worked.

File: src/harness/fourslashImpl.ts (hunk 1 above)
Comment (MariaSolOs) [reply src/harness/fourslashImpl.ts:845]: Yes. If the file is defined, the span definitely is.

**Review by jakebailey — APPROVED** [review]

**Pull request merged**